rust/otap-dataflow/crates/telemetry/src/opentelemetry_client/meter_provider/prometheus_exporter_provider.rs

```diff
@@ -0,0 +1,244 @@
+
+        Ok((sdk_meter_builder, Some(tokio_runtime)))
+    }
+
+    fn validate_config(prometheus_config: &PrometheusExporterConfig) -> Result<(), Error> {
+        let endpoint = format!("{}:{}", prometheus_config.host, prometheus_config.port);
+        let _parsed_endpoint = endpoint.parse::<SocketAddr>().map_err(|e| {
+            Error::ConfigurationError(format!("Invalid Prometheus bind address: {}", e))
+        })?;
+
+        let path = &prometheus_config.path;
+        if !path.starts_with('/') {
+            return Err(Error::ConfigurationError(
+                "Prometheus metrics path must start with '/'".to_string(),
+            ));
+        }
+
+        Ok(())
+    }
+
+    fn start_async_prometheus_server(
+        registry: Registry,
+        runtime: tokio::runtime::Runtime,
+        prometheus_config: &PrometheusExporterConfig,
+    ) -> Result<tokio::runtime::Runtime, Error> {
+        let endpoint = format!("{}:{}", prometheus_config.host, prometheus_config.port);
+        let path = prometheus_config.path.clone();
+        let _server_handle = runtime.spawn(async move {
+            Self::start_prometheus_server(registry, &endpoint, &path)
+                .await
+                .map_err(|e| {
+                    Error::ConfigurationError(format!("Failed to start Prometheus server: {}", e))
+                })
+        });
+        Ok(runtime)
+    }
+
+    async fn start_prometheus_server(
+        registry: Registry,
+        endpoint: &str,
+        path: &str,
+    ) -> Result<(), Error> {
+        let addr: SocketAddr = endpoint.parse().map_err(|e| {
+            Error::ConfigurationError(format!("Invalid Prometheus bind address: {}", e))
+        })?;
+
+        let listener = tokio::net::TcpListener::bind(addr).await.map_err(|e| {
+            Error::ConfigurationError(format!("Failed to bind Prometheus server: {}", e))
+        })?;
+
+        let app = Router::new()
+            .merge(Self::routes(path))
+            .with_state(registry.clone());
+
+        axum::serve(listener, app)
+            .await
+            .map_err(|e| Error::ConfigurationError(format!("Prometheus server failed: {}", e)))?;
+
+        Ok(())
+    }
+
+    /// Define the routes for the Prometheus exporter.
+    fn routes(path: &str) -> Router<Registry> {
+        Router::new().route(path, get(Self::get_metrics))
+    }
+
+    /// Handler for the `/metrics` endpoint.
+    async fn get_metrics(State(registry): State<Registry>) -> Result<Response, StatusCode> {
+        let encoder = TextEncoder::new();
+        let metric_families = registry.gather();
+        let mut buffer = Vec::new();
+        encoder
+            .encode(&metric_families, &mut buffer)
+            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+
+        let response = Response::builder()
+            .status(StatusCode::OK)
+            .header("Content-Type", "text/plain; version=0.0.4; charset=utf-8")
```

Comment:
we may have to support openmetrics format in the future. Not something for this PR.